### PR TITLE
[url_launcher] Update readme for URL schemes on iOS

### PR DESCRIPTION
**Pull request by TahaTesser**
## Description

Add instructions for iOS, for targeting iOS 9.0 and higher. 
Note:
All new projects 1.22+ targets iOS 9.0 by default 
Screenshot 2020-11-06 at 4 50 28 PM
97959488-6d67bd00-1dea-11eb-9cdf-a06ea55738d5

## Related Issues

Related to https://github.com/flutter/flutter/issues/65936

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: url_launcher
 description: Flutter plugin for launching a URL on Android and iOS. Supports
   web, phone, SMS, and email schemes.
 homepage: https://github.com/flutter/plugins/tree/master/packages/url_launcher/url_launcher
-version: 5.7.10
+version: 5.7.10+1
```
Comment (hamdikahloun) [inline packages/url_launcher/url_launcher/pubspec.yaml:5]: The next Minor version must be `5.7.11`

Comment (TahaTesser) [reply packages/url_launcher/url_launcher/pubspec.yaml:5]: Update doc can be just bumped by +1  
Example
https://pub.dev/packages/image_picker/changelog#06710
https://pub.dev/packages/url_launcher/changelog#0421 
https://pub.dev/packages/google_maps_flutter/changelog#052113

Also example from my last PR
https://github.com/flutter/plugins/pull/2932/files

Comment (hamdikahloun) [reply packages/url_launcher/url_launcher/pubspec.yaml:5]: @TahaTesser 

For package versions > 1.0.0, we shouldn't do +z .

Details in https://dart.dev/tools/pub/versioning#semantic-versions

```
Running version check for changed packages
packages/url_launcher/url_launcher/pubspec.yaml incorrectly updated version.
HEAD: 5.7.10+1, master: 5.7.10.
Allowed versions: {6.0.0: NextVersionType.BREAKING_MAJOR, 6.0.0-nullsafety: NextVersionType.MAJOR_NULLSAFETY_PRE_RELEASE, 5.8.0-nullsafety: NextVersionType.MINOR_NULLSAFETY_PRE_RELEASE, 5.8.0: NextVersionType.MINOR, 5.7.11: NextVersionType.PATCH}
```

Comment (TahaTesser) [reply packages/url_launcher/url_launcher/pubspec.yaml]: That's interesting,  made the changes required. 
Thank you

File: packages/url_launcher/url_launcher/README.md [hunk 2]
```diff
@@ -8,6 +8,23 @@ iOS, Android, web, Windows, macOS, and Linux.
...
+## Installation
+
+### iOS 
+
+When targeting iOS 9.0+, new default from Flutter 1.22+.
+Add the following keys to your _Info.plist_ file, located in `<project root>/ios/Runner/Info.plist`:
+
+```
+<key>LSApplicationQueriesSchemes</key>
+<array>
```
Comment (hamdikahloun) [inline packages/url_launcher/url_launcher/README.md:20]: Does this cover other [Schemes](https://developer.apple.com/library/archive/featuredarticles/iPhoneURLScheme_Reference/Introduction/Introduction.html) like `mailto`, `tel` and `sms`, or we must add new items for this property?

Comment (TahaTesser) [reply packages/url_launcher/url_launcher/README.md:20]: Those schemes are supported without adding this so no new items are required

**Comment by TahaTesser** [comment]
CC: @cyanglaz

**Comment by TahaTesser** [comment]
@hamdikahloun 
Is there anything we need to merge this?
Thank you

**Comment by hamdikahloun** [comment]
hi @TahaTesser 
Last time i checked it, the branch has conflicts that must be resolved .
Please fix the conflicts before merging .
Thank you

**Review by hamdikahloun — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
@hamdikahloun 
Just rebased and bumped version, update changelog
I've removed the `When targeting iOS 9.0+, new default from Flutter 1.22+.` line since this version of stable has been around for months
Can you please review that current updated PR?
Thank you

**Comment by hamdikahloun** [comment]
> @hamdikahloun
> Just rebased and bumped version, update changelog
> I've removed the `When targeting iOS 9.0+, new default from Flutter 1.22+.` line since this version of stable has been around for months
> Can you please review that current updated PR?
> Thank you

LGTM

**Review by darshankawar — APPROVED** [review]

**Comment by darshankawar** [comment]
LGTM

File: packages/url_launcher/url_launcher/README.md [hunk 3]
```diff
@@ -8,6 +8,22 @@ iOS, Android, web, Windows, macOS, and Linux.
 ## Usage
 To use this plugin, add `url_launcher` as a [dependency in your pubspec.yaml file](https://flutter.dev/platform-plugins/).
 
+## Installation
+
+### iOS 
+
+Add the following keys to your _Info.plist_ file, located in `<project root>/ios/Runner/Info.plist`:
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:15]: This is not an accurate summary of the requirement, and may still cause confusion. The correct description is that the scheme of any URL they pass to `canLaunch` should be listed, and the README should say that. You can give the below as an *example*, but you should make clear that it's an example.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/url_launcher/url_launcher/README.md [hunk 4]
```diff
@@ -8,6 +8,22 @@ iOS, Android, web, Windows, macOS, and Linux.
 ## Usage
 To use this plugin, add `url_launcher` as a [dependency in your pubspec.yaml file](https://flutter.dev/platform-plugins/).
 
+## Installation
+
+### iOS 
+Add scheme for any URL passed on `canLaunch` to your _Info.plist_ file.
```
Comment (darshankawar) [inline packages/url_launcher/url_launcher/README.md:14]: May be "Add any URL scheme passed to `canLaunch` in your _Info.plist_ file."

File: packages/url_launcher/url_launcher/README.md (hunk 3 above)
Comment (TahaTesser) [reply packages/url_launcher/url_launcher/README.md]: Made the changes,  thanks!

**Comment by darshankawar** [comment]
LGTM

File: packages/url_launcher/url_launcher/README.md [hunk 5]
```diff
@@ -8,6 +8,22 @@ iOS, Android, web, Windows, macOS, and Linux.
 ## Usage
 To use this plugin, add `url_launcher` as a [dependency in your pubspec.yaml file](https://flutter.dev/platform-plugins/).
 
+## Installation
+
+### iOS 
+Add any URL scheme passed to canLaunch in your Info.plist file.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:14]: Nits:
- Add `s around canLaunch
- s/in your/as `LSApplicationQueriesSchemes` entries in your/

File: packages/url_launcher/url_launcher/README.md [hunk 6]
```diff
@@ -8,6 +8,22 @@ iOS, Android, web, Windows, macOS, and Linux.
...
+Example:  
+```
+<key>LSApplicationQueriesSchemes</key>
+<array>
+  <string>https</string>
+  <string>http</string>
+</array>
+```
+
+See [canOpenURL](https://developer.apple.com/documentation/uikit/uiapplication/1622952-canopenurl) for more details.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/README.md:25]: Nit: the standard way to refer to an OjbC method without context includes the class, and the full signature (which includes the `-` and the `:`), like: `-[UIApplication canOpenURL:]` 
```
See [`-[UIApplication canOpenURL:]`](...) for more details.
```

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits

**Comment by TahaTesser** [comment]
@stuartmorgan 
Sorry for the delay, made the changes.

**Review by stuartmorgan — APPROVED** [review]
LGTM!

**Comment by stuartmorgan** [comment]
I've restarted the failed test, since it was a network issue.

**Pull request merged**